### DISCUSSION TITLE
Remote image processing

### DISCUSSION THREAD

**Discussion by RoySalisbury** [aaronwmorris/indi-allsky#1259] [General]
What would be the best way (if its even possible) to setup a system so that the "imaging" PC (a Raspberry Pi) is only used for the actual image capture, but the rest of the processing (or as much as possible) is running on another system?

It would seem that if I could use a different "processing" system for the heavy lifting of image processing (panorama, keogram, startrails, stacking, etc).  I could setup this processing system on my local Proxmox host into whatever OS is needed and give it much higher performance (CPU, GPU, memory, etc).

I see that there are different docker containers for different things, but does not really seem built around separating the pieces between systems.

**Comment by Vertikar** [comment]
I had a crack at this a while ago, and if you're running a camera natively supported by indi, like an ZWO or Playerone for example, it should be relatively smooth.

In short, you would just run indi-allsky in a docker container on the processing system, point it to the imaging system, which you'd install INDI server on, and it should work.

I was trying with a libcamera camera, and I ran into some other complicated issues that unfortunately I never got around, and life got in the way of picking it up again.

You don't need to run full indi-allsky on the imaging PC, you only need INDI server, which you can configure relatively easy using [INDI Web Manager](https://github.com/knro/indiwebmanager). That said I did have a few issues with that crashing, and taking down the underlying INDI server, so a better way may be to just install INDI server and configure it from the CLI with its config files.

There doesn't seem to be a recently updated docker container for INDI server that I can find in a quick google, if you wanted to go down that path, but that may just over complicate it anyway,

@aaronwmorris may have tried this in the past using indi-allsky, or know of a better way to do it.

**Reply by RoySalisbury** [reply]
So what I tried last night was to setup a VM of Debian 12 and went through the entire indi-allsky setup on it (including having to build all the indi drivers).  Once it was installed I configured the imaging Raspi Pi to disable all the processing logic in the UI.  Then on the processing server I disabled the indi server and set the camaera config to point to the IP address of the Raspberry Pi.

This works so far.  Now I will work on trimming it all down.  The processing is now MUCH faster for the images (even transferring them via WiFi from the Raspi Pi).  I can process a full HD image from the camera in about 0.2 seconds (before it was about 7 seconds).

I wonder if there is a way to install indi-allsky without having to build all the indi hardware.

Roy

**Reply by aaronwmorris** [reply]
The indiserver component can most definitely be separated from the indi-allsky install.  The connection to the indiserver is already network enabled, it is just configured to communicate with "localhost" by default.  However, it is entirely supported to run the indiserver on a remote system with the camera, say on a Raspi, and indi-allsky can run on a more powerful system.

In the case of the remote indi-allsky install, you would only need to build the INDI core component.  The 3rd party drivers only needed where the camera is connected.
